[#30] Update README file

lookeypl opened the pull request:
This change resolves #30 - adds new information to README file, performs some formatting for better readability and updates DEPENDENCIES section due to #33.

Verification:  
- Read entire README file, check for spelling errors, verify if all that is written makes sense and will be a good introductory file for others.  
- Check new [wiki page](https://github.com/nfprojects/nfengine/wiki/Introduction), which is a Markdown-formatted version of README file.

lookeypl committed:
Add information about nfEngine workflow

This commit adds detailed information about workflow in nfEngine:
* Section FORMATTING is updated with basic formatting information
* New sections are created:
    - ISSUE TRACKING
    - WORKFLOW AND BRANCHING IN NFENGINE
    - COMMIT MESSAGE FORMAT
    - PULL REQUEST DESCRIPTION FORMAT

lookeypl committed:
Align README file to 100 chars per line

One part of README file exceeded 100 chars. This commit resolves this
problem.

lookeypl committed:
Slightly reformat README file

This commit performs slight reformmating to README file:
* To make separate sections of README more visible, indented each section
  title by 4 spaces.
* Added CONTENTS section to inform user where specific sections are
  located in the document.

lookeypl committed:
Realign README file to 85 chars per line

On GitHub README is displayed with 85 chars per line formatting. This
commit fixes README to be aligned to 85 chars/line, not 100.

Additionally, some minor changes were addressed.

lookeypl committed:
Update Visual Studio entry in DEPENDENCIES section

Witek902 commented:
Looks good, but there is one more thing: could you rename "README" file so it is automatically parsed by github and displayed properly?

lookeypl committed:
Convert README to README.md

This commit converts README file to Markdown format. Additionally,
extension is changed to .md, so that GitHub can properly parse the file.

Commit:
Update README.md

lookeypl commented:
Updated
